src/backend/kvm/thread.rs

```diff
@@ -77,77 +77,132 @@ impl<P: KeepPersonality> Thread<P> {
                 let block_virt = self.keep.write().unwrap().sallyports[block_nr]
                     .take()
                     .unwrap();
 
                 // If some other thread tried to use the same block, the above unwrap would have panicked.
-                let block = unsafe { &mut *block_virt.as_mut_ptr::<Block>() };
-
-                // To avoid clashing of rep and req in the union, clone the request
-                let req = unsafe { block.msg.req };
-
-                let ret = match i64::from(req.num) {
-                    SYS_ENARX_BALLOON_MEMORY => {
-                        let rep = self.balloon(&req).into();
-                        block.msg.rep = rep;
-                        Ok(Command::Continue)
-                    }
-
-                    SYS_ENARX_MEM_INFO => {
-                        block.msg.rep = self.meminfo(block).into();
-                        Ok(Command::Continue)
+                let block: Block = unsafe {
+                    std::slice::from_raw_parts_mut(
+                        block_virt.as_mut_ptr::<usize>(),
+                        self.keep.read().unwrap().sallyport_block_size as usize
+                            / size_of::<usize>(),
+                    )
+                }
+                .into();
+
+                for item in block {
+                    match item {
+                        Item::Gdbcall(_gdbcall, _data) => {
+                            #[cfg(feature = "gdb")]
+                            unsafe {
+                                execute_gdb(
+                                    _gdbcall,
+                                    _data,
+                                    &mut self.gdb_fd,
+                                    _gdblisten.as_ref().unwrap(),
+                                )
+                                .map_err(io::Error::from_raw_os_error)
+                                .context("execute_gdb")?;
+                            }
+                        }
+
+                        Item::Enarxcall(enarxcall, data) => {
+                            sallyport::host::execute(
+                                self.kvm_enarxcall(enarxcall, data)?.into_iter(),
+                            )
+                            .map_err(io::Error::from_raw_os_error)
+                            .context("sallyport::host::execute")?;
+                        }
+
+                        // Catch exit and exit_group for a clean shutdown
+                        Item::Syscall(syscall, ..)
+                            if (syscall.num == libc::SYS_exit as usize
+                                || syscall.num == libc::SYS_exit_group as usize) =>
+                        {
+                            if cfg!(feature = "dbg") {
+                                dbg!(&syscall);
+                            }
+                            return Ok(Command::Exit(syscall.argv[0] as _));
+                        }
+
+                        Item::Syscall(ref _syscall, ..) => {
```

Comment:
drop the `_`

Reply:
if feature `dbg` is not used, `_syscall` is unused

Reply:
I can think of two ways to work around that:
1. `cfg!` macro
2. having something like `let _ = syscall` to ensure it's used

Reply:
Although the best solution is probably to just disable the lint here?